### PR TITLE
ENH+BF: Do not fail reporting if curdir is gone, do not rmdir the directory upon failed clone attempt

### PR DESCRIPTION
**Pull request by yarikoptic**
Addresses #2787

### CODE REVIEW

File: datalad/support/path.py [hunk 1]
```diff
@@ -48,3 +51,18 @@ def wrapped(*args, **kwargs):
...
+    """A helper which would not fail if p is relative and we are in non-existing directory
+
+    It will rely on getpwd, which would rely on $PWD env variable to report
+    the path.  Desired for improved resilience during e.g. reporting as in
+    https://github.com/datalad/datalad/issues/2787
+    """
+    try:
+        return abspath(p)
+    except OSError as exc:
+        if "No such" in str(exc) and not isabs(p):
```
Comment (kyleam) [inline datalad/support/path.py]: Eh I dislike relying on matching a particular string in the exception message, though I know there are already lots of case where we do it in the codebase.  In this case, couldn't you `getpwd()` and raise unless it *doesn't* exist and `not isabs(p)`?

Comment (yarikoptic) [reply datalad/support/path.py]: Ok, will do shortly

Comment (yarikoptic) [reply datalad/support/path.py]: Although it would make reraising original exception more cumbersome iirc

**Comment by yarikoptic** [comment]
ok, approved is approved! ;)

**Pull request merged**